localstack/services/awslambda/resource_providers/aws_lambda_alias.py

```diff
@@ -0,0 +1,149 @@
+# LocalStack Resource Provider Scaffolding v2
+from __future__ import annotations
+
+from pathlib import Path
+from typing import Optional, Type, TypedDict
+
+import localstack.services.cloudformation.provider_utils as util
+from localstack.services.cloudformation.resource_provider import (
+    CloudFormationResourceProviderPlugin,
+    OperationStatus,
+    ProgressEvent,
+    ResourceProvider,
+    ResourceRequest,
+)
+
+
+class LambdaAliasProperties(TypedDict):
+    FunctionName: Optional[str]
+    FunctionVersion: Optional[str]
+    Name: Optional[str]
+    Description: Optional[str]
+    Id: Optional[str]
+    ProvisionedConcurrencyConfig: Optional[ProvisionedConcurrencyConfiguration]
+    RoutingConfig: Optional[AliasRoutingConfiguration]
+
+
+class ProvisionedConcurrencyConfiguration(TypedDict):
+    ProvisionedConcurrentExecutions: Optional[int]
+
+
+class VersionWeight(TypedDict):
+    FunctionVersion: Optional[str]
+    FunctionWeight: Optional[float]
+
+
+class AliasRoutingConfiguration(TypedDict):
+    AdditionalVersionWeights: Optional[list[VersionWeight]]
+
+
+REPEATED_INVOCATION = "repeated_invocation"
+
+
+class LambdaAliasProvider(ResourceProvider[LambdaAliasProperties]):
+
+    TYPE = "AWS::Lambda::Alias"  # Autogenerated. Don't change
+    SCHEMA = util.get_schema_path(Path(__file__))  # Autogenerated. Don't change
+
+    def create(
+        self,
+        request: ResourceRequest[LambdaAliasProperties],
+    ) -> ProgressEvent[LambdaAliasProperties]:
+        """
+        Create a new resource.
+
+        Primary identifier fields:
+          - /properties/Id
+
+        Required properties:
+          - FunctionName
+          - FunctionVersion
+          - Name
+
+        Create-only properties:
+          - /properties/Name
+          - /properties/FunctionName
+
+        Read-only properties:
+          - /properties/Id
+
+
+
+        """
+        model = request.desired_state
+        lambda_ = request.aws_client_factory.lambda_
+
+        create_params = {
+            k: v
+            for k, v in model.items()
+            if k in ["FunctionName", "FunctionVersion", "Name", "Description", "RoutingConfig"]
+        }
```

Comment:
you can use the functions available in the imported `util` package which should have a helper for this (if you want)